pkg/server/plugin/zmq/broker.go

```diff
@@ -194,54 +197,80 @@ func newWorker(address []byte) pworker {
 	}
 }
 
-type workerQueue []pworker
+type pworkers []pworker
+
+// workerQueue is a last tick fist out queue.
+// A worker need to register itself using Add before it can tick.
+// Ticking of an non-registered worker will be no-ops.
+type workerQueue struct {
+	pworkers
+	addresses map[string]bool
+}
+
+func newWorkerQueue() workerQueue {
+	return workerQueue{
+		pworkers{},
+		map[string]bool{},
+	}
+}
 
 func (q workerQueue) Len() int {
-	return len(q)
+	return len(q.pworkers)
 }
 
 func (q *workerQueue) Next() []byte {
-	workers := *q
+	workers := q.pworkers
 	worker := workers[len(workers)-1]
-	*q = workers[:len(workers)-1]
+	q.pworkers = workers[:len(workers)-1]
 	return worker.address
 }
 
-func (q *workerQueue) Ready(worker pworker) {
-	workers := *q
+func (q *workerQueue) Add(worker pworker) {
+	q.addresses[string(worker.address)] = true
+	err := q.Tick(worker)
+	if err == nil {
+		return
+	}
+}
+
+func (q *workerQueue) Tick(worker pworker) error {
+	if _, ok := q.addresses[string(worker.address)]; !ok {
```

Comment:
`workerQueue.pworkers` Only holy the idle worker.  That is the queue for consuming order. It son't store all previously registered worker. So the state is not duplicated.

Hence, if we make `Add` only register worker, but not push to the queue. It will take next round heartbeat to push itself to worker.